near-contract-standards/src/fungible_token/storage_impl.rs

```diff
@@ -0,0 +1,102 @@
+        &mut self,
+        force: Option<bool>,
+    ) -> Option<(AccountId, Balance)> {
+        assert_one_yocto();
+        let account_id = env::predecessor_account_id();
+        let force = force.unwrap_or(false);
+        if let Some(balance) = self.accounts.get(&account_id) {
+            if balance == 0 || force {
+                self.accounts.remove(&account_id);
+                self.total_supply -= balance;
+                Promise::new(account_id.clone()).transfer(self.storage_balance_bounds().min.0 + 1);
+                Some((account_id, balance))
+            } else {
+                env::panic(b"Can't unregister the account with the positive balance without force")
+            }
+        } else {
+            env::log(b"The account is not registered");
+            None
+        }
+    }
+
+    fn internal_storage_balance_of(&self, account_id: &AccountId) -> Option<StorageBalance> {
+        if self.accounts.contains_key(account_id) {
+            Some(StorageBalance { total: self.storage_balance_bounds().min, available: 0.into() })
+        } else {
+            None
+        }
+    }
+}
+
+impl StorageManagement for FungibleToken {
+    // `registration_only` doesn't affect the implementation for vanilla fungible token.
+    #[allow(unused_variables)]
+    fn storage_deposit(
+        &mut self,
+        account_id: Option<ValidAccountId>,
+        registration_only: Option<bool>,
+    ) -> StorageBalance {
+        let amount: Balance = env::attached_deposit();
+        let account_id =
+            account_id.map(|a| a.into()).unwrap_or_else(|| env::predecessor_account_id());
+        if self.accounts.contains_key(&account_id) {
+            log!("The account is already registered, refunding the deposit");
+            if amount > 0 {
+                Promise::new(env::predecessor_account_id()).transfer(amount);
+            }
+        } else {
+            let min_balance = self.storage_balance_bounds().min.0;
+            if amount < min_balance {
+                env::panic(b"The attached deposit is less than the mimimum storage balance");
+            }
+
+            self.internal_register_account(&account_id);
+            let refund = amount - min_balance;
+            if refund > 0 {
+                Promise::new(env::predecessor_account_id()).transfer(refund);
+            }
+        }
+        self.internal_storage_balance_of(&account_id).unwrap()
+    }
+
+    fn storage_withdraw(&mut self, amount: Option<U128>) -> StorageBalance {
+        if let Some(storage_balance) =
+            self.internal_storage_balance_of(&env::predecessor_account_id())
+        {
+            match amount {
+                Some(amount) if amount.0 > 0 => {
+                    env::panic(b"The amount is greater than the available storage balance");
+                }
+                _ => storage_balance,
+            }
+        } else {
+            env::panic(b"The account does not exist");
```

Comment:
"account does not exist" sounds to me like "invalid account id". Can we make this "The account is not registered"?

Reply:
Can we make errors more parsable by software in general?
E.g. format!("ERR_NO_USER:{}", user)

Or have enum of these errors or something?

We have this thing everywhere, which is really hard later to show any meaningful errors in frontends.

Reply:
I think this should come through the event NEP: https://github.com/near/NEPs/issues/130
It looks like the goal is to use Borsh for events, should we try something new unparsable here?

Reply:
I'd like to continue this discussion about error message later and affirm that this isn't blocking for the PR. 🙏🏼